core/src/main/java/jenkins/util/Timer.java

```diff
@@ -39,7 +41,8 @@ public static synchronized ScheduledExecutorService get() {
         if (executorService == null) {
             // corePoolSize is set to 10, but will only be created if needed.
             // ScheduledThreadPoolExecutor "acts as a fixed-sized pool using corePoolSize threads"
-            executorService =  new ErrorLoggingScheduledThreadPoolExecutor(10, new NamingThreadFactory(new DaemonThreadFactory(), "jenkins.util.Timer"));
+            // TODO consider also wrapping in ContextResettingExecutorService
+            executorService = new ImpersonatingScheduledExecutorService(new ErrorLoggingScheduledThreadPoolExecutor(10, new NamingThreadFactory(new DaemonThreadFactory(), "jenkins.util.Timer")), ACL.SYSTEM);
```

Comment:
This fixes all sorts of background tasks to run as `SYSTEM`, the way you probably thought they already were, but apparently were not. In particular, `TryRepeatedly`, used to restore the state of a running Pipeline build after restart.